web/yo/app/scripts/controllers/gene.js

```diff
@@ -1629,94 +1616,44 @@ angular.module('oncokbApp')
                 }
                 return uuids;
             }
-
-            function removeInModel(mutationIndex, tumorTypeIndex, therapyCategoryIndex, therapyIndex) {
-                var _index = -1;
-                console.log(mutationIndex, tumorTypeIndex, therapyCategoryIndex, therapyIndex);
-                if (angular.isNumber(mutationIndex)) {
-                    if (!_.isUndefined(mutationIndex)) {
-                        if (_.isUndefined(tumorTypeIndex)) {
-                            _index = Number(angular.copy(mutationIndex));
-                            $scope.gene.mutations.remove(_index);
-                            delete $scope.geneStatus[_index];
-                            $scope.geneStatus = migrateGeneStatusPosition($scope.geneStatus, _index);
-                        } else if (!_.isUndefined(therapyCategoryIndex) && !_.isUndefined(therapyIndex)) {
-                            _index = Number(angular.copy(therapyIndex));
-                            var treatments = $scope.gene.mutations.get(mutationIndex).tumors.get(tumorTypeIndex).TI.get(therapyCategoryIndex).treatments;
-                            treatments.remove(_index);
-                            // this needs to be put back after fix the treatment index bug
-                            delete $scope.geneStatus[mutationIndex][tumorTypeIndex][therapyCategoryIndex][_index];
-                            $scope.geneStatus[mutationIndex][tumorTypeIndex][therapyCategoryIndex] = migrateGeneStatusPosition($scope.geneStatus[mutationIndex][tumorTypeIndex][therapyCategoryIndex], _index);
```

Comment:
I didn't see any place in the new code you update the `geneStauts`. Don't you need to update this object as well?